Dependent crates with lesser versions are being used by rust-skeptic

6 | use rand::distributions::Alphanumeric;

  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Alphanumeric` in `distributions`

error[E0599]: no method named `sample` found for type `rand::ThreadRng` in the current scope

  --> /tmp/rust-skeptic.lV2m9aQedOqB/test.rs:13:29

   |

13 |     *p = (0..5).map(|_| rng.sample(&Alphanumeric)).collect()

   |                             ^^^^^^

5 | use rand::distributions::{Distribution, Uniform};

  |                           ^^^^^^^^^^^^  ^^^^^^^ no `Uniform` in `distributions`

  |                           |

  |                           no `Distribution` in `distributions`

warning: unused import: `Rng`

 --> /tmp/rust-skeptic.NnKO8W2SxQw2/test.rs:4:24

  |

4 | use rand::{thread_rng, Rng};

  |                        ^^^

  |

  = note: `#[warn(unused_imports)]` on by default

error[E0599]: no method named `sample_iter` found for type `rand::ThreadRng` in the current scope

 --> /tmp/rust-skeptic.NnKO8W2SxQw2/test.rs:9:10

  |

9 |         .sample_iter(&Alphanumeric)

  |

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,6 +24,8 @@ matrix:
     - rust: stable
       os: linux
       env: CONTENT_TESTS=1 CONTENT_TESTS_LINKS=1
+    - rust: stable 
+      os: osx
 
 addons:
   apt:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ data-encoding = "2.1.0"
 env_logger = "0.5"
 error-chain = "0.12"
 flate2 = "1.0"
-glob = "0.2"
+glob = "0.3"
 image = "0.20"
 lazy_static = "1.0"
 log = "0.4"
@@ -63,9 +63,9 @@ walkdir = "2.0"
 syslog = "5.0"
 
 [build-dependencies]
-skeptic = "0.13"
+skeptic = { git = 'https://github.com/andygauge/rust-skeptic'}
 walkdir = "2.0"
 
 [dev-dependencies]
-skeptic = "0.13"
+skeptic = { git = 'https://github.com/andygauge/rust-skeptic'}
 walkdir = "2.0"

--- a/src/algorithms/randomness/rand-custom.md
+++ b/src/algorithms/randomness/rand-custom.md
@@ -5,7 +5,7 @@
 Randomly generates a tuple `(i32, bool, f64)` and variable of user defined type `Point`.
 Implements the [`Distribution`] trait on type Point for [`Standard`] in order to allow random generation.
 
-```rust,edition2018,ignore
+```rust,edition2018
 use rand::Rng;
 use rand::distributions::{Distribution, Standard};
 

--- a/src/algorithms/randomness/rand-dist.md
+++ b/src/algorithms/randomness/rand-dist.md
@@ -14,9 +14,10 @@ An example using the [`Normal`] distribution is shown below.
 
 ```rust,edition2018,ignore
 use rand_distr::{Distribution, Normal, NormalError};
+use rand::thread_rng;
 
 fn main() -> Result<(), NormalError> {
-    let mut rng = rand::thread_rng();
+    let mut rng = thread_rng();
     let normal = Normal::new(2.0, 3.0)?;
     let v = normal.sample(&mut rng);
     println!("{} is from a N(2, 9) distribution", v);

--- a/src/algorithms/randomness/rand-passwd.md
+++ b/src/algorithms/randomness/rand-passwd.md
@@ -5,7 +5,7 @@
 Randomly generates a string of given length ASCII characters in the range `A-Z,
 a-z, 0-9`, with [`Alphanumeric`] sample.
 
-```rust,edition2018,ignore
+```rust,edition2018
 use rand::{thread_rng, Rng};
 use rand::distributions::Alphanumeric;
 

--- a/src/algorithms/randomness/rand-range.md
+++ b/src/algorithms/randomness/rand-range.md
@@ -4,7 +4,7 @@
 
 Generates a random value within half-open `[0, 10)` range (not including `10`) with [`Rng::gen_range`].
 
-```rust,edition2018,ignore
+```rust,edition2018
 use rand::Rng;
 
 fn main() {
@@ -18,7 +18,7 @@ fn main() {
 This has the same effect, but may be faster when repeatedly generating numbers
 in the same range.
 
-```rust,edition2018,ignore
+```rust,edition2018
 
 use rand::distributions::{Distribution, Uniform};
 

--- a/src/concurrency/parallel/rayon-parallel-sort.md
+++ b/src/concurrency/parallel/rayon-parallel-sort.md
@@ -9,7 +9,7 @@ values in parallel.  Although [multiple options]
 exist to sort an enumerable data type, [`par_sort_unstable`]
 is usually faster than [stable sorting] algorithms.
 
-```rust,edition2018,ignore
+```rust,edition2018
 
 use rand::{Rng, thread_rng};
 use rand::distributions::Alphanumeric;

--- a/src/concurrency/parallel/rayon-thumbnails.md
+++ b/src/concurrency/parallel/rayon-thumbnails.md
@@ -29,7 +29,7 @@ use rayon::prelude::*;
 
 fn main() -> Result<()> {
     let options: MatchOptions = Default::default();
-    let files: Vec<_> = glob_with("*.jpg", &options)?
+    let files: Vec<_> = glob_with("*.jpg", options)?
         .filter_map(|x| x.ok())
         .collect();
 

--- a/src/cryptography/encryption/pbkdf2.md
+++ b/src/cryptography/encryption/pbkdf2.md
@@ -9,7 +9,7 @@ function [`pbkdf2::derive`].  Verifies the hash is correct with
 [`SecureRandom::fill`], which fills the salt byte array with
 securely generated random numbers.
 
-```rust,edition2018,ignore
+```rust,edition2018
 
 use data_encoding::HEXUPPER;
 use ring::error::Unspecified;

--- a/src/development_tools/build_tools/cc-bundled-static.md
+++ b/src/development_tools/build_tools/cc-bundled-static.md
@@ -56,24 +56,24 @@ void greet(const char* name) {
 ### `src/main.rs`
 
 ```rust,edition2018,ignore
+use error_chain::error_chain;
 use std::ffi::CString;
 use std::os::raw::c_char;
-#
-# error_chain! {
-#     foreign_links {
-#         NulError(::std::ffi::NulError);
-#         Io(::std::io::Error);
-#     }
-# }
-#
-# fn prompt(s: &str) -> Result<String> {
-#     use std::io::Write;
-#     print!("{}", s);
-#     std::io::stdout().flush()?;
-#     let mut input = String::new();
-#     std::io::stdin().read_line(&mut input)?;
-#     Ok(input.trim().to_string())
-# }
+
+error_chain! {
+    foreign_links {
+        NulError(::std::ffi::NulError);
+        Io(::std::io::Error);
+    }
+}
+fn prompt(s: &str) -> Result<String> {
+    use std::io::Write;
+    print!("{}", s);
+    std::io::stdout().flush()?;
+    let mut input = String::new();
+    std::io::stdin().read_line(&mut input)?;
+    Ok(input.trim().to_string())
+}
 
 extern {
     fn hello();

--- a/src/encoding/string/percent-encode.md
+++ b/src/encoding/string/percent-encode.md
@@ -6,7 +6,7 @@ Encode an input string with [percent-encoding] using the [`utf8_percent_encode`]
 function from the `percent-encoding` crate. Then decode using the [`percent_decode`]
 function.
 
-```rust,edition2018,ignore
+```rust,edition2018
 use percent_encoding::{utf8_percent_encode, percent_decode, AsciiSet, CONTROLS};
 use std::str::Utf8Error;
 

--- a/src/file/dir/ignore-case.md
+++ b/src/file/dir/ignore-case.md
@@ -7,16 +7,15 @@ Find all image files in the `/media/` directory matching the `img_[0-9]*.png` pa
 A custom [`MatchOptions`] struct is passed to the [`glob_with`] function making the glob pattern case insensitive while keeping the other options [`Default`].
 
 ```rust,edition2018,no_run
-# use error_chain::error_chain;
-
+use error_chain::error_chain;
 use glob::{glob_with, MatchOptions};
-#
-# error_chain! {
-#     foreign_links {
-#         Glob(glob::GlobError);
-#         Pattern(glob::PatternError);
-#     }
-# }
+
+error_chain! {
+    foreign_links {
+        Glob(glob::GlobError);
+        Pattern(glob::PatternError);
+    }
+}
 
 fn main() -> Result<()> {
     let options = MatchOptions {
@@ -24,7 +23,7 @@ fn main() -> Result<()> {
         ..Default::default()
     };
 
-    for entry in glob_with("/media/img_[0-9]*.png", &options)? {
+    for entry in glob_with("/media/img_[0-9]*.png", options)? {
         println!("{}", entry?.display());
     }
 

--- a/src/web/clients/api/rate-limited.md
+++ b/src/web/clients/api/rate-limited.md
@@ -7,7 +7,7 @@ handle remote server errors.  This example uses the [`hyper::header!`] macro
 to parse the response header and checks for [`reqwest::StatusCode::Forbidden`].
 If the response exceeds the rate limit, the example waits and retries.
 
-```rust,edition2018,no_run,ignore
+```rust,edition2018,no_run
 # use error_chain::error_chain;
 
 use std::time::{Duration, UNIX_EPOCH};

--- a/src/web/clients/requests/header.md
+++ b/src/web/clients/requests/header.md
@@ -14,7 +14,7 @@ with [`RequestBuilder::header`] then makes the request with
 The request targets <http://httpbin.org/headers> service which responds with
 a JSON dict containing all request headers for easy verification.
 
-```rust,edition2018,no_run,ignore
+```rust,edition2018,no_run
 # use error_chain::error_chain;
 use serde::Deserialize;
 

--- a/src/web/scraping/broken.md
+++ b/src/web/scraping/broken.md
@@ -17,7 +17,6 @@ use reqwest::StatusCode;
 use select::document::Document;
 use select::predicate::Name;
 use std::collections::HashSet;
-use tokio::stream::{self, StreamExt};
 use url::{Position, Url};
 
 error_chain! {

